apps/angular/1-projection/src/app/ui/card/card.component.ts

```diff
@@ -1,58 +1,36 @@
-import { NgOptimizedImage } from '@angular/common';
-import { Component, inject, input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
-import { ListItemComponent } from '../list-item/list-item.component';
+import { NgTemplateOutlet } from '@angular/common';
+import { Component, input, output, TemplateRef } from '@angular/core';
 
 @Component({
   selector: 'app-card',
   template: `
-    <div
-      class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass()">
-      @if (type() === CardType.TEACHER) {
-        <img ngSrc="assets/img/teacher.png" width="200" height="200" />
-      }
-      @if (type() === CardType.STUDENT) {
-        <img ngSrc="assets/img/student.webp" width="200" height="200" />
-      }
+    <ng-content select="img"></ng-content>
 
-      <section>
-        @for (item of list(); track item) {
-          <app-list-item
-            [name]="item.firstName"
-            [id]="item.id"
-            [type]="type()"></app-list-item>
-        }
-      </section>
+    <section>
+      @for (item of list(); track item) {
+        <ng-container
+          [ngTemplateOutlet]="itemTemplate()"
+          [ngTemplateOutletContext]="{ $implicit: item }"></ng-container>
+      }
+    </section>
 
-      <button
-        class="rounded-sm border border-blue-500 bg-blue-300 p-2"
-        (click)="addNewItem()">
-        Add
-      </button>
-    </div>
+    <button
+      class="rounded-sm border border-blue-500 bg-blue-300 p-2"
+      (click)="addNewItem()">
+      Add
+    </button>
   `,
-  imports: [ListItemComponent, NgOptimizedImage],
+  host: {
+    class: 'border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3',
+  },
+  imports: [NgTemplateOutlet],
 })
-export class CardComponent {
-  private teacherStore = inject(TeacherStore);
-  private studentStore = inject(StudentStore);
-
-  readonly list = input<any[] | null>(null);
-  readonly type = input.required<CardType>();
-  readonly customClass = input('');
-
-  CardType = CardType;
+export class CardComponent<T> {
+  readonly list = input<T[] | null>(null);
+  readonly itemTemplate = input<TemplateRef<any> | null>(null);
```

Comment:
you can do it this way or you can use `contentChild` to get the template from the parent